pkg/vm/run_context.go

```diff
@@ -9,3 +13,61 @@ type RunContext struct {
 	Ap memory.Relocatable
 	Fp memory.Relocatable
 }
+
+func (run_context RunContext) ComputeDstAddr(instruction Instruction) (memory.Relocatable, error) {
+	var base_addr memory.Relocatable
+	switch instruction.DstReg {
+	case AP:
+		base_addr = run_context.Ap
+	case FP:
+		base_addr = run_context.Fp
+	}
+
+	if instruction.OffOp0 < 0 {
+		return base_addr.SubUint(uint(math.Abs(float64(instruction.OffDst))))
+	} else {
+		return base_addr.AddUint(uint(instruction.OffDst))
+	}
+
+}
+
+func (run_context RunContext) ComputeOp0Addr(instruction Instruction) (memory.Relocatable, error) {
+	var base_addr memory.Relocatable
+	switch instruction.Op0Reg {
+	case AP:
+		base_addr = run_context.Ap
+	case FP:
+		base_addr = run_context.Fp
+	}
+
+	if instruction.OffOp1 < 0 {
+		return base_addr.SubUint(uint(math.Abs(float64(instruction.OffOp0))))
+	} else {
+		return base_addr.AddUint(uint(instruction.OffOp0))
+	}
+}
+
+func (run_context RunContext) ComputeOp1Addr(instruction Instruction, op0 memory.MaybeRelocatable) (memory.Relocatable, error) {
```

Comment:
op0 is not necessarily known at this point, we can receive a *memory.MaybeRelocatable and check if its not nil